### ISSUE TITLE
Unsupported major.minor version 52.0 exception

### ISSUE THREAD

**Issue by janskarvall** [PavlikPolivka/GitLabProjects#27]
I just, for the first time, tried to install GitLab Projects in Android Studio. After restart of Android Studio I get an alert pop-up: Plugin Error, GitLab Projects threw an uncaught PluginException. Disable Plugin.

Looking in the `idea.log` I find:

```
2016-04-12 20:00:06,248 [  62652]  ERROR - ctionSystem.DefaultActionGroup - Android Studio 2.0  Build #AI-143.2739321 
2016-04-12 20:00:06,248 [  62652]  ERROR - ctionSystem.DefaultActionGroup - JDK: 1.7.0_95 
2016-04-12 20:00:06,248 [  62652]  ERROR - ctionSystem.DefaultActionGroup - VM: OpenJDK 64-Bit Server VM 
2016-04-12 20:00:06,248 [  62652]  ERROR - ctionSystem.DefaultActionGroup - Vendor: Oracle Corporation 
2016-04-12 20:00:06,248 [  62652]  ERROR - ctionSystem.DefaultActionGroup - OS: Linux 
2016-04-12 20:00:06,248 [  62652]  ERROR - ctionSystem.DefaultActionGroup - Last Action: ShowSettings 
2016-04-12 20:00:06,302 [  62706]  ERROR - ctionSystem.DefaultActionGroup - cannot create class "com.ppolivka.gitlabprojects.share.GitLabShareAction" [Plugin: com.ppolivka.gitlabprojects] 
com.intellij.diagnostic.PluginException: cannot create class "com.ppolivka.gitlabprojects.share.GitLabShareAction" [Plugin: com.ppolivka.gitlabprojects]
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:177)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convert(ActionManagerImpl.java:515)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getActionImpl(ActionManagerImpl.java:495)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.getAction(ActionManagerImpl.java:488)
        at com.intellij.openapi.actionSystem.DefaultActionGroup.unStub(DefaultActionGroup.java:354)
        at com.intellij.openapi.actionSystem.DefaultActionGroup.getChildren(DefaultActionGroup.java:312)
        at com.intellij.openapi.actionSystem.impl.Utils.hasChildrenWithState(Utils.java:225)
        at com.intellij.openapi.actionSystem.impl.Utils.hasVisibleChildren(Utils.java:205)
        at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:153)
        at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:86)
        at com.intellij.openapi.actionSystem.impl.Utils.fillMenu(Utils.java:280)
        at com.intellij.openapi.actionSystem.impl.ActionMenu.fillMenu(ActionMenu.java:314)
        at com.intellij.openapi.actionSystem.impl.ActionMenu.access$600(ActionMenu.java:50)
        at com.intellij.openapi.actionSystem.impl.ActionMenu$MenuListenerImpl.menuSelected(ActionMenu.java:268)
        at javax.swing.JMenu.fireMenuSelected(JMenu.java:1043)
        at javax.swing.JMenu$MenuChangeListener.stateChanged(JMenu.java:1122)
        at javax.swing.DefaultButtonModel.fireStateChanged(DefaultButtonModel.java:349)
        at javax.swing.DefaultButtonModel.setSelected(DefaultButtonModel.java:232)
        at javax.swing.JMenu.setSelected(JMenu.java:311)
        at javax.swing.JMenu.menuSelectionChanged(JMenu.java:1190)
        at com.intellij.openapi.actionSystem.impl.ActionMenu.menuSelectionChanged(ActionMenu.java:240)
        at javax.swing.MenuSelectionManager.setSelectedPath(MenuSelectionManager.java:124)
        at javax.swing.plaf.basic.BasicMenuUI$Handler.mousePressed(BasicMenuUI.java:450)
        at java.awt.Component.processMouseEvent(Component.java:6513)
        at javax.swing.JComponent.processMouseEvent(JComponent.java:3312)
        at java.awt.Component.processEvent(Component.java:6281)
        at java.awt.Container.processEvent(Container.java:2229)
        at java.awt.Component.dispatchEventImpl(Component.java:4872)
        at java.awt.Container.dispatchEventImpl(Container.java:2287)
        at java.awt.Component.dispatchEvent(Component.java:4698)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4832)
        at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4489)
        at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4422)
        at java.awt.Container.dispatchEventImpl(Container.java:2273)
        at java.awt.Window.dispatchEventImpl(Window.java:2719)
        at java.awt.Component.dispatchEvent(Component.java:4698)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:747)
        at java.awt.EventQueue.access$300(EventQueue.java:103)
        at java.awt.EventQueue$3.run(EventQueue.java:706)
        at java.awt.EventQueue$3.run(EventQueue.java:704)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:77)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
        at java.awt.EventQueue$4.run(EventQueue.java:720)
        at java.awt.EventQueue$4.run(EventQueue.java:718)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:77)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:717)
        at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:866)
        at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:650)
        at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:381)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:242)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:161)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:150)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:146)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:138)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:91)
Caused by: com.intellij.diagnostic.PluginException: com/ppolivka/gitlabprojects/share/GitLabShareAction : Unsupported major.minor version 52.0 [Plugin: com.ppolivka.gitlabprojects]
        at com.intellij.ide.plugins.cl.PluginClassLoader.loadClassInsideSelf(PluginClassLoader.java:130)
        at com.intellij.ide.plugins.cl.PluginClassLoader.tryLoadingClass(PluginClassLoader.java:77)
        at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:66)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:278)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.convertStub(ActionManagerImpl.java:153)
        ... 56 more
```

**Comment by PavlikPolivka** [comment]
I recently switched to new IntelliJ SDK that is now on Java8. You Android Studio runs on java 7. So there might some problems with new SDK. I need to investigate this more, but everything seems to be working with my android studio on jdk 1.8 na Windows. 
Unfortunately my best bet here is that I will end up forcing plugin to Java 8 only IDEs (All Jetbrains IDEs are forced to run jdk1.8 or higher starting from version 16).
https://intellij-support.jetbrains.com/hc/en-us/articles/206544879-Selecting-the-JDK-version-the-IDE-will-run-under
Plugin does not need this, but it is difficult do develop for older IDE versions than one you have installed.

Quick fix for you is to run Android Studio on JDK 1.8 and force language level down to 1.7 because my understanding here is that java 8 is not yet supported for android development.

**Comment by PavlikPolivka** [comment]
I can change minimal; required version of a IDE to 144 (Java 8 required), Android Studio currently does not have any build in this version. 
If you want to use Java 7 compatible version you can download it from here: https://plugins.jetbrains.com/plugin/download?pr=&updateId=23065 and install it manually.

I will try to compile next version of the plugin (1.3.2) under Java 7 and make it compatible with android studio and IntelliJ 15 and lower platforms. 
All other development will be done for versions 144 and higher, 1.3.2 will still show in repository for IDEs not compatible with new versions (mainly android studio).

**Comment by janskarvall** [comment]
I upgraded to Android Studio 2.0 just before installing GitLab Projects.
I had not noted, but Android Studio 2.0 actually requires Java Development Kit (JDK) 8, [See System Requirement here](http://developer.android.com/sdk/index.html).

So, maybe, you should not bother.

I'm currently using OpenJDK 7 on Ubuntu 14.04 LTS. OpenJDK 8 is not available on 14.04, but I think it will be on 16.04. And of course I could install the Oracle version.

**Comment by PavlikPolivka** [comment]
New version of plugin will not install for IDEs that are not running on Java 8
